### ISSUE TITLE
Wrong error message for interface in `.js` file

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#35968]
In a `.js` file:

```js
export interface Stuff {
}
```

**Expected**: `Interfaces can only be used within TypeScript files.`

**Actual**: `Interface declaration cannot have 'implements' clause.`
